lib/cretonne/src/ir/dfg.rs

```diff
@@ -562,6 +562,22 @@ impl DataFlowGraph {
                         })
     }
 
+    /// Append a new value argument to the branch jump to a given destination.
+    ///
+    /// Panics if the instruction does not branch.
+    pub fn append_branch_argument(&mut self, inst: Inst, new_arg: Value) {
```

Comment:
Should we generalize this to an `append_inst_arg()` function? I could see it being useful for calls too, perhaps.